### DISCUSSION TITLE
Next.js server side rendering warning, need to provide <IdProvider>

### DISCUSSION THREAD

**Discussion by mstibbard** [supabase/supabase#2368] [Questions]
I'm currently trying out supabase auth and Next.js and it's been fantastic so far! However, when running `yarn dev` I occasionally get the following warning in the terminal.

`When server rendering, you must wrap your application in an <IdProvider> to ensure consistent ids are generated between the client and server.`

Is this ID not provided by the Auth.UserContextProvider?

pages/_app.tsx
```
import '../styles/globals.css';
import { supabase } from '../utils/supabaseClient';
import { ThemeProvider } from 'next-themes';
import type { AppProps } from 'next/app';
import { Auth } from '@supabase/ui';

export default function MyApp({ Component, pageProps }: AppProps) {
  return (
    <Auth.UserContextProvider supabaseClient={supabase}>
      <ThemeProvider attribute='class'>
        <Component {...pageProps} />
      </ThemeProvider>
    </Auth.UserContextProvider>
  );
}
```

and 

../utils/supabaseClient
```
import { createClient } from '@supabase/supabase-js';

const supabaseUrl: string = process.env.NEXT_PUBLIC_SUPABASE_URL!;
const supabaseAnonKey: string = process.env.NEXT_PUBLIC_SUPABASE_ANON_KEY!;

export const supabase = createClient(supabaseUrl, supabaseAnonKey);
```

**Comment by dshukertjr** [comment]
Hi @mstibbard!

Are you using `Radix` to create your app? Maybe the error message is coming from `Radix` and not Supabase or Next.js. 

https://www.radix-ui.com/docs/primitives/utilities/id-provider

It seems like you just need to wrap your components with `IdProvider` the way described [here](https://www.radix-ui.com/docs/primitives/overview/server-side-rendering).

**Reply by mstibbard** [reply]
I am not using `Radix`. It also doesn't always pop up... I'm trying to consistently reproduce it but no luck yet.

**Reply by dshukertjr** [reply]
Hmm, that is odd. Sorry that I couldn't hep you out...

**Reply by MildTomato** [reply]
This is possibly because we use Radix inside of Supabase UI. But we don't use it in the Auth component - so it's a bit strange you're seeing this.

Maybe try moving the ThemeProvider to be the parent component.
